File: verein-web/src/types/organization.ts
export type OrganizationType = 'Landesverband' | 'Region' | 'Verein';
export type FederationCode = 'DITIB';

export interface OrganizationDto {
  id: number;
  name: string;
  orgType: OrganizationType | string;
  parentOrganizationId?: number | null;
  federationCode?: FederationCode | string | null;
  aktiv?: boolean;
  deletedFlag?: boolean;
}

export interface OrganizationCreateDto {
  name: string;
  orgType: OrganizationType | string;
  parentOrganizationId?: number | null;
  federationCode?: FederationCode | string | null;
  aktiv?: boolean;
}

export interface OrganizationUpdateDto {
  name?: string;
  orgType?: OrganizationType | string;
  parentOrganizationId?: number | null;
  federationCode?: FederationCode | string | null;
  aktiv?: boolean;
}

export interface TreeNodeDto {
  id: number;
  name: string;
  orgType: OrganizationType | string;
  deletedFlag?: boolean;
  children: TreeNodeDto[];
}

export interface PathNodeDto {
  id: number;
  name: string;
  orgType: OrganizationType | string;
}

File: verein-api/Domain/Entities/Organization.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VereinsApi.Domain.Entities;

/// <summary>
/// Organization entity representing hierarchical organizational units
/// </summary>
[Table("Organization")]
public class Organization : AuditableEntity
{
    /// <summary>
    /// Display name of the organization unit
    /// </summary>
    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Organization type (Landesverband, Region, Verein)
    /// </summary>
    [Required]
    [MaxLength(20)]
    public string OrgType { get; set; } = string.Empty;

    /// <summary>
    /// Parent organization identifier (self-referencing FK)
    /// </summary>
    public int? ParentOrganizationId { get; set; }

    /// <summary>
    /// Federation code (DITIB)
    /// </summary>
    [MaxLength(20)]
    public string? FederationCode { get; set; }

    /// <summary>
    /// Parent organization navigation
    /// </summary>
    public virtual Organization? ParentOrganization { get; set; }

    /// <summary>
    /// Child organizations navigation
    /// </summary>
    public virtual ICollection<Organization> Children { get; set; } = new List<Organization>();
}
